include/highfive/xtensor.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "bits/H5Inspector_decl.hpp"
+#include "H5Exception.hpp"
+
+#include <xtensor/xtensor.hpp>
+#include <xtensor/xarray.hpp>
+#include <xtensor/xadapt.hpp>
+
+namespace HighFive {
+namespace details {
+
+template <class XTensorType, bool IsConstExprRowMajor>
+struct xtensor_inspector {
+    using type = XTensorType;
+    using value_type = typename type::value_type;
+    using base_type = typename inspector<value_type>::base_type;
+    using hdf5_type = base_type;
+
+    // TODO prevent non-trivial elements.
+
+    static constexpr size_t ndim = type::rank;
+    static constexpr size_t recursive_ndim = ndim + inspector<value_type>::recursive_ndim;
+    static constexpr bool is_trivially_copyable = IsConstExprRowMajor &&
+                                                  std::is_trivially_copyable<value_type>::value &&
+                                                  inspector<value_type>::is_trivially_copyable;
+
+    static std::vector<size_t> getDimensions(const type& val) {
+        std::array<size_t, ndim> shape = val.shape();
+        std::vector<size_t> sizes(shape.begin(), shape.end());
+        return sizes;
+    }
+
+    static void prepare(type& val, const std::vector<size_t>& dims) {
+        auto shape = std::array<size_t, ndim>{};
+        std::copy(dims.begin(), dims.begin() + ndim, shape.begin());
+        val.resize(shape);
+    }
```

Comment:
no recursivity here?

Reply:
None of it works with non-trivial elements. I should check if they even allow anything non-trivial.

Reply:
We're now asserting the fact. I propose we revisit, if we ever need to support them.